homeassistant/components/thermopro/button.py

```diff
@@ -0,0 +1,169 @@
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the demo button platform."""
+    assert entry.unique_id is not None
+
+    set_datetime_entity = None
+
+    @callback
+    def _async_on_data_updated(
+        data: ThermoProBluetoothDeviceData,
+        service_info: BluetoothServiceInfoBleak,
+        update: SensorUpdate,
+    ) -> None:
+        nonlocal set_datetime_entity
+        _LOGGER.debug(
+            "got update data=%s update=%s service_info=%s", data, update, service_info
+        )
+
+        assert entry.unique_id is not None
+        assert None in update.devices
+        assert update.devices[None].model is not None
+
+        if update.devices[None].model not in ("TP358", "TP393"):
+            return
+
+        if not set_datetime_entity:
+            name = update.devices[None].name
+            assert name is not None
+            set_datetime_entity = ThermoProDateTimeButtonEntity(
+                entry_id=entry.entry_id,
+                address=entry.unique_id,
+                device_name=name,
+                description=DATETIME_UPDATE,
+            )
+
+            async_add_entities(
+                [
+                    set_datetime_entity,
+                ]
+            )
+
+        if service_info.connectable and not set_datetime_entity.available:
+            _LOGGER.debug("sending availability '%s' for %s", True, entry.entry_id)
+            async_dispatcher_send(
+                hass,
+                f"{SIGNAL_AVAILABILITY_UPDATED}_{entry.entry_id}_button_datetime",
+                True,
+            )
+
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, f"{SIGNAL_DATA_UPDATED}_{entry.entry_id}", _async_on_data_updated
+        )
+    )
+
+    @callback
+    def _async_on_unavailable(service_info: BluetoothServiceInfoBleak) -> None:
+        _LOGGER.debug("service info unavailable %s", service_info)
+        if set_datetime_entity:
+            _LOGGER.debug("sending availability '%s' for %s", False, entry.entry_id)
+            async_dispatcher_send(
+                hass,
+                f"{SIGNAL_AVAILABILITY_UPDATED}_{entry.entry_id}_button_datetime",
+                False,
+            )
+
+    entry.async_on_unload(
+        async_track_unavailable(
+            hass, _async_on_unavailable, entry.unique_id, connectable=True
+        )
+    )
+
+
+class ThermoProDateTimeButtonEntity(
+    ButtonEntity,
+):
+    """Representation of a ThermoProDateTime button entity."""
+
+    def __init__(
+        self,
+        entry_id: str,
+        address: str,
+        device_name: str,
+        description: ButtonEntityDescription,
+    ) -> None:
+        """Initialize the thermopro datetime button entity."""
+        self.address = address
+        self.entity_description = description
+        self.entry_id = entry_id
+        self._attr_unique_id = f"{device_name}-{description.key}"
```

Comment:
```suggestion
        self._attr_unique_id = f"{address}-{description.key}"
```

We should use the stable identifier for the unique id